contracts/BatchExchangeViewer.sol

```diff
@@ -146,9 +146,9 @@ contract BatchExchangeViewer {
             uint16 nextPageUserOffset
         )
     {
-        elements = new bytes(maxPageSize * AUCTION_ELEMENT_WIDTH);
+        elements = new bytes(uint256(maxPageSize) * INDEXED_AUCTION_ELEMENT_WIDTH);
         setLength(elements, 0);
-        bytes memory unfiltered = new bytes(maxPageSize * AUCTION_ELEMENT_WIDTH);
+        bytes memory unfiltered = new bytes(uint256(maxPageSize) * AUCTION_ELEMENT_WIDTH);
```

Comment:
Still wondering about this recently introduced usage of `uint256` cast. I didn't see any other types change... why weren't we using this before if it is necessary now?

Reply:
They are necessary when maxPageSize * AUCTION_ELEMENT_WIDTH > 2**16, thus at page sizes >585 (I don't think we have been using more than 500 page sizes until now). They were necessary before and not having them was a bug.